[package] minizip-ng/4.1.0: minizip recipe always links against system zlib-ng if present

### Description

An application that depends on `minizip-ng` may fail to link with errors like "undefined reference to `zng_deflateInit2'".

The minizip-ng recipe declares `zlib` as a dependency. However, the underlying cmake build script calls `find_package(ZLIB-NG)` before `find_package(ZLIB)`; it prefers to link to zlib-ng if it is found. This seems like a bug - it should _always_ link to the conan-provided zlib library.

The cmake build script provides an option to control this behavior. It seems like the conan minizip-ng recipe should somehow set `MZ_ZLIB_FLAVOR=zlib` to force linking against the conan-provided zlib.

### Package and Environment Details

* Package Name/Version: **minizip-ng/4.1.0**
* docker image: **archlinux:multilib-devel-20260412.0.513370**
* Operating System+version: **Linux (Arch)**
* Compiler+version: **GCC 15.2.1**
* Conan version: **conan 2.27.1**
* Python version: **Python 3.14.4**


### Conan profile

[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=gnu17
compiler.libcxx=libstdc++11
compiler.version=15
os=Linux

### Steps to reproduce

```bash
# run following commands in a docker container
docker run --rm -it archlinux:multilib-devel-20260412.0.513370

# install system packages + conan
pacman -Syy && pacman -S --noconfirm base-devel gcc cmake python python-pip jq
pip install conan --upgrade --break-system-packages

# install minizip-ng system package
pacman -S --noconfirm minizip-ng

# download and build minizip-ng conan pacakge
conan profile detect
conan install --requires=minizip-ng/4.1.0 -r=conancenter --build='*'

# list unresolved symbols in static library
PKG_ID=`conan list 'minizip-ng/*:*' --format=json | jq -r '."Local Cache"."minizip-ng/4.1.0".revisions | to_entries | .[0].value.packages | keys[0]'`
PKG_PATH=`conan cache path minizip-ng/4.1.0:$PKG_ID`
echo "Symbols in libminizip-ng that come from zlib-ng:"
nm -u $PKG_PATH/lib/libminizip-ng.a | grep zng_

# output of `nm ... | grep ...` SHOULD be empty
# but is not be b/c minizip-ng erronously expects to link to zlib-ng
```

### Logs

```
# ... MANY lines out output from setup and package compilation ...
Symbols in libminizip-ng that come from zlib-ng:
[root@0fba1c703a09 /]# nm -u $PKG_PATH/lib/libminizip-ng.a | grep zng_
                 U zng_crc32
                 U zng_deflate
                 U zng_deflateEnd
                 U zng_deflateInit2
                 U zng_inflate
                 U zng_inflateEnd
                 U zng_inflateInit2
```

--- a/recipes/minizip-ng/all/conanfile.py
+++ b/recipes/minizip-ng/all/conanfile.py
@@ -43,10 +43,6 @@ class MinizipNgConan(ConanFile):
         "with_libcomp": True,
     }
 
-    @property
-    def _is_clang_cl(self):
-        return self.settings.os == "Windows" and self.settings.compiler == "clang"
-
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
@@ -92,6 +88,7 @@ class MinizipNgConan(ConanFile):
         tc.cache_variables["MZ_FETCH_LIBS"] = False
         tc.cache_variables["MZ_COMPAT"] = self.options.mz_compatibility
         tc.cache_variables["MZ_ZLIB"] = self.options.get_safe("with_zlib", False)
+        tc.cache_variables["MZ_ZLIB_FLAVOR"] = "zlib"
         tc.cache_variables["MZ_BZIP2"] = self.options.with_bzip2
         tc.cache_variables["MZ_LZMA"] = self.options.with_lzma
         tc.cache_variables["MZ_ZSTD"] = self.options.with_zstd
@@ -125,7 +122,6 @@ class MinizipNgConan(ConanFile):
         self.cpp_info.set_property("cmake_target_name", "MINIZIP::minizip")
         self.cpp_info.set_property("pkg_config_name", "minizip")
 
-        # TODO: back to global scope in conan v2 once cmake_find_package_* generators removed
         suffix = "" if self.options.mz_compatibility else "-ng"
         self.cpp_info.components["minizip"].libs = [f"minizip{suffix}"]
         if self.options.with_lzma:

